doc/sphinx/generate_doc.py

```diff
@@ -85,14 +91,41 @@ def make_title(title):
     "azure.mgmt.resource.subscriptions",
 ]
 
-def generate_doc(config_path, output_directory = "./ref/", project_pattern=None):
-    multiapi_found_apiversion = {}
-
-    rst_path_template = os.path.join(output_directory, '{}.rst')
-    rst_namespace_template = os.path.join(output_directory, '{}.{}.rst')
 
+def get_updated_config(config_path: str, package_root: str) -> Dict[str, Dict[str, typing.Union[str, List[str]]]]:
     with Path(config_path).open() as config_fd:
         config = json.load(config_fd)
+
+    namespace_folders = os.path.basename(package_root).split("-")
+    package_name = "-".join(namespace_folders)
+    namespace = ".".join(namespace_folders)
+
+    api_directory = os.path.join(package_root, *namespace_folders)
+    glob_path = os.path.join(api_directory, "v20*/")
+    valid_versions = glob.glob(glob_path)
+
+    for version_path in valid_versions:
+        api_version = os.path.relpath(version_path, start=api_directory)
+        full_namespace = namespace + f".{api_version}"
+
+        if config[package_name]:
+            ns = config[package_name]["namespaces"]
```

Comment:
Do we need any error handling for this?

Reply:
Know what, yeah. I should add protection.